gcc/rust/checks/errors/borrowck/rust-bir-builder-expr-stmt.cc

```diff
@@ -281,7 +281,46 @@ ExprStmtBuilder::visit (HIR::CallExpr &expr)
 
 void
 ExprStmtBuilder::visit (HIR::MethodCallExpr &expr)
-{}
+{
+  // TODO : need ExprStmtBuilder::visit_expr (HIR::PathExprSegment)
+  PlaceId fn = visit_expr (expr.get_method_name ());
+
+  std::vector<PlaceId> arguments = visit_list (expr.get_arguments ());
+
+  const auto fn_type = ctx.place_db[fn].tyty->as<const TyTy::FnType> ();
+
+  // compile `self` and handle autoref
+  PlaceId self = visit_expr (*expr.get_receiver ());
+  auto ty = ctx.place_db[self].tyty;
+  if (ty->get_kind () == TyTy::REF)
```

Comment:
This seems to be inverted. Autoref needs to make reference to values that are not references.

Reply:
Autoderef may be needed here. What you want to achieve is to have exactly one reference (in terms of levels).